src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -577,7 +577,7 @@ private static void checkAllele(String allele, boolean isRef, int lineNo) {
             System.err.println(String.format("Allele detected with length %d exceeding max size %d at approximately line %d, likely resulting in degraded VCF processing performance", allele.length(), MAX_ALLELE_SIZE_BEFORE_WARNING, lineNo));
         }
 
-        if ( isSymbolicAllele(allele) ) {
+        if (Allele.wouldBeSymbolicAllele(allele.getBytes())) {
```

Comment:
This check is done before calling the `Allele` constructor, but it appears to be ensuring that we create a valid Allele. Is there a reason why this code isn't part of the `Allele` constructor? We would have to wrap the constructor call in a try/catch to get a message printed out with the VCF line number, though.

Reply:
Yeah, there's a big redundant section of allele checking code here that I believe is redundant with the constructor.  I wanted to make this as small a change set as possible though.  I started a follow up branch that cleans up some of this stuff but I figured I'd get this one in before that.